src/components/TokenBalance.tsx

```diff
@@ -87,20 +97,26 @@ function TokenBalance({
         <Image source={{ uri: tokensWithUsdValue[0].imageUrl }} style={styles.tokenImg} />
         <View style={styles.column}>
           <Text style={style} testID={'TotalTokenBalance'}>
-            {localCurrencySymbol}
-            {totalTokenBalanceLocal?.toFormat(2) ?? '-'}
-          </Text>
-          <Text style={styles.tokenBalance}>
-            {formatValueToDisplay(tokenBalance)} {tokensWithUsdValue[0].symbol}
+            {(overrideHideBalance || !hideBalance) && localCurrencySymbol}
+            {hideBalance && !overrideHideBalance
+              ? 'XX' + decimalSeparator + 'XX'
+              : totalTokenBalanceLocal?.toFormat(2) ?? '-'}
           </Text>
+          {!hideBalance && (
+            <Text style={styles.tokenBalance}>
+              {formatValueToDisplay(tokenBalance)} {tokensWithUsdValue[0].symbol}
+            </Text>
+          )}
         </View>
       </View>
     )
   } else {
     return (
       <Text style={style} testID={'TotalTokenBalance'}>
-        {localCurrencySymbol}
-        {totalBalanceLocal?.toFormat(2) ?? new BigNumber(0).toFormat(2)}
+        {(overrideHideBalance || !hideBalance) && localCurrencySymbol}
+        {hideBalance && !overrideHideBalance
+          ? 'XX' + decimalSeparator + 'XX'
+          : totalBalanceLocal?.toFormat(2) ?? new BigNumber(0).toFormat(2)}
```

Comment:
can we make this DRY, seems like you could have a var outside the if's e.g., 
```
const balanceDisplay = hideBalance ? `XX${decimalSeparator}XX` : totalBalanceLocal?.toFormat(2)
```
and then do 
```
balanceDisplay ?? `-`
```
and 
```
balanceDisplay ?? new BigNumber(0).toFormat(2)
```
in the two places

Also it's better to use template literals over string concatenation